### PR TITLE
feat(pet-148): persistent scan-history back-pages (on-disk sink + cursor + paged view)

### PR DESCRIPTION
**Pull request by ziomancer**
## Summary
- Real back-pages for the console scan history: an append-only, rotation-bounded **on-disk sink** retains rows beyond the in-memory 500-entry ring, an additive **`before` cursor** on `get_scan_history` seeks past the ring into the sink, and a **paged Observability view** walks older history. The ring stays a hard 500 cap (the fast path); depth comes from the sink, not a bigger deque or a config knob.
- Inherits the ring's PII-at-rest discipline verbatim (no `matched_text`, no normalized-text preview at rest) and PET-139's keyed-HMAC attestation under a **distinct** domain subkey (`b"petasos/scan-history/v1"`); the on-path append is fail-open and stays off the scan latency budget.
- New module `petasos/console/_history.py` is the newest member of the cross-process file-sink family (sibling of `_events.py`); the one deliberate divergence is that rotation **reclaims** the oldest `.rot` (a queryable store whose oldest rows are meant to age out), where the enforcement spool refuses to drop undrained events.

## Layered defense (how the records stay in lockstep)
The sink append lives only on the single `_record_scan` chokepoint that both record paths already route through, so "a scan happened" advances the **ring**, **`scans_total`**, and the **on-disk sink** together (PET-131 invariant). The append is fail-open (`append_history_row` returns `False`, never raises) with a one-shot `PETASOS_HISTORY_SINK_UNWRITABLE` tripwire so a dead sink is observable, not silent. Rotation runs after **every** append path (`run_scan` *and* the enforcement drain), so the sink is bounded in both standalone (background tailer) and embedded (no tailer) modes regardless of reads.

## Files changed

| File | Change |
|------|--------|
| `petasos/console/_history.py` | New — fail-open append, fail-safe key-ordered reverse-scan reader, reclaiming single-`.rot` rotation, domain-separated HMAC subkey |
| `petasos/console/server.py` | Sink append on `_record_scan` (+ one-shot tripwire); additive `before` cursor on `get_scan_history` (`next_before`/`older_truncated`); `_maybe_rotate_history` after every append path; `scan_history_capacity` in `get_health`; `_SCAN_HISTORY_RING_CAPACITY` constant; full-hex playground `scan_id` |
| `petasos/console/hermes/plugin_api.py` | Thread the `before` query param to the handler |
| `petasos/console/static/petasos.js` | `before` arg on `getScanHistory`; pure paging seams (`historyPagingView`/`historyPageLabel`); Older/Newer controls + retention-honest empty state; `/* @ring-cap */` drift marker |
| `tests/test_console_scan_history.py` | New — 16 regression tests (paging/ordering/dedup, bounded growth both modes, PII-at-rest, restart survival, attestation, fail-open, cursor edge cases, latency, per-profile, drift guard) |
| `tests/js/scan-history-paging.test.mjs` | New — node:vm harness over the paging seams |
| `tests/conftest.py` | Autouse `_isolate_history_sink` fixture (redirects the sink to a temp path) |
| `tests/test_console_handlers.py` | `test_get_scan_history_empty` updated for the additive response keys |

## Test plan
- [x] `C:\python310\python.exe -m pytest --deselect "tests/test_console_validation.py::test_default_ignorable_rejected"` — 1962 passed, 41 skipped, 1 deselected, 1 xfailed (full output: docs/specs/TODO/PET-148.test-output.txt, gitignored)
- [x] `ruff check .` — clean
- [x] `ruff format --check .` — 143 files already formatted
- [x] `mypy --strict .` — no issues in 143 source files
- [x] `node --test tests/js/*.test.mjs` — 173 passed (incl. the new paging seam suite)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## New Features
* Added persistent, size-bounded scan-history back-pages stored on disk, enabling cursor-based **Older/Newer** navigation beyond the live in-memory window.
* Updated the scan-history API and console UI to accept an optional `before` cursor and return pagination metadata (`next_before`, `older_truncated`).
* Added optional per-record cryptographic attestation for persisted history; tampered records are shown as unverifiable.

## Bug Fixes
* Hardened profile-name option building to prevent `__proto__`-related dedup bypass issues.

## Tests
* Expanded console paging, rotation/bounding, restart survival, PII minimization, tamper detection, and fail-open/fail-safe coverage (including new JS paging seam tests and an auto-isolating history sink fixture).

### CODE REVIEW

File: petasos/console/server.py [hunk 1]
```diff
@@ -435,6 +524,56 @@ def _record_scan(self, summary: dict[str, Any]) -> None:
...
+        reads. The embedded Hermes plugin starts no background tailer, so checking rotation
+        only on the drain would leave a never-polled embedded sink unbounded under sustained
+        playground scans. Acquires only _history_lock (sequentially with, never nested
+        inside, _enforcement_lock); a long O(retained) page read holding that lock blocks
+        only a later rotation, never the enforcement drain. Never raises."""
+        async with self._history_lock:
+            try:
+                if _history.history_size() > _history._HISTORY_CAP_BYTES:
+                    _history.rotate_history()
+            except Exception:
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/server.py]: Resolved in 364cc00. `_maybe_rotate_history` now captures `rotate_history()`'s return and emits a one-shot `PETASOS_HISTORY_ROTATION_STALLED` WARNING when rotation is refused while over cap, mirroring the existing `PETASOS_HISTORY_SINK_UNWRITABLE` guard, so bounded-retention failure is observable rather than silent. Added a regression test (`test_rotation_stall_warns_once`) pinning the warn-once gate.

File: petasos/console/static/petasos.js [hunk 2]
```diff
@@ -1627,12 +1757,20 @@
...
       Pet.api.getScanHistory(500).then(function (d) {
         if (Pet.auth.on401(d)) return; // PET-129 D3: first statement, before the shape-guarded seed-merge
         // startFallbackPolling response-shape guard, NOT the bare !d.error check:
         // _req never rejects on HTTP error (it resolves an error envelope), and a
         // 200 {} body lacking .entries would make the merge throw on .scan_id.
         if (!d.error && d.entries && Array.isArray(d.entries)) {
+          // PET-148: the server's next_before on the seed (limit 500) is the cursor at the
+          // oldest seeded row — the boundary the first "Older" click pages past, server-minted
+          // so the client never mints a float-repr-fragile token (edge F-4).
+          Pet.state.historyHeadCursor = d.next_before || null;
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/static/petasos.js:1785]: Confirmed real, deferred to PET-152. The cursor is seed-time-only and goes stale as the ring evicts, opening an unreachable slice after >500 scans within a single mount (rows stay in the on-disk sink, so it is a UI-navigation gap, not data loss). A correct fix re-mints a server-side cursor on the first "Older"-from-head click and touches the unit-tested `historyPagingView` plus new JS coverage, so per repo convention it is routed to a brief + spec-cycle rather than an ad-hoc review edit. Tracking: PET-152.
